src/pngimage.cpp

```diff
@@ -589,9 +589,9 @@ namespace Exiv2 {
                     ExifParser::encode(blob, littleEndian, exifData_);
                     if (blob.size() > 0)
                     {
-                        static const char exifHeader[] = { 0x45, 0x78, 0x69, 0x66, 0x00, 0x00 };
-                        std::string rawExif =   std::string(exifHeader, 6)
-                                              + std::string((const char*)&blob[0], blob.size());
+                        const std::string exifHeader{"Exif\0\0", 6};
```

Comment:
suggestion: make this an `std::array`, afaik string allocates on the heap, as it cannot take ownership of a `const char*`.

Reply:
Is this what you are suggesting?

```cpp
const std::array<char, 6> exifHeader{'E','x','i','f',0,0};
```

Reply:
No, rather:
```cpp
const std::array<byte, 6> exifHeader{ 0x45, 0x78, 0x69, 0x66, 0x00, 0x00 };
```
Will do.

Reply:
Why no `std::array` here?

Reply:
`PngChunk::makeMetadataChunk` expects string. So why should I first create array, just to convert it to string?